### PR TITLE
Implement test_auto_remount_with_subpath

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/1719

### CODE REVIEW

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -2035,39 +2036,78 @@ def test_auto_remount_with_subpath():
...
-    11. Kill the engine instance manager pod on `node-1`
-    12. In a 2 minutes retry loop:
-       Exec into the `pod-1`, run `ls /mnt/html`.
-       Verify the file `test_data.txt` exists.
-
-    13. kill `pod-1`.
-    14. In a 2 minutes retry loop:
-       Exec into the `pod-1`, run `ls /mnt/html`.
-       Verify the file `test_data.txt` exists.
+       Repeat these steps for 3 times.
```
Comment (shuo-wu) [inline manager/integration/tests/test_ha.py:2046]: This line is a little bit confusing. How about the followings instead:
```
4. Delete the statefulset replica instance manager pod. This action simulates a network disconnection. 
5. Wait for volume `healthy`, then verify the file checksum.
6. Repeat step #4~#5 for 3 times.
```

File: manager/integration/tests/test_ha.py [hunk 2]
```diff
@@ -2035,39 +2036,78 @@ def test_auto_remount_with_subpath():
...
+        1. Wait for volume `healthy`.
+        2. Verify the file checksum.
+    5. Update `numberOfReplicas` to 3.
+        1. Wait for replicas rebuilding finishes.
+    6. Delete one of the statefulset engine instance manager pod.
+        1. Verify the file checksum.
+    7. Delete `pod-1`.
+        1. Verify the file checksum.
```
Comment (shuo-wu) [inline manager/integration/tests/test_ha.py:2054]: Wait for pod recreation and volume remount. Then verify the file checksum.

File: manager/integration/tests/test_ha.py [hunk 3]
```diff
@@ -2035,39 +2036,78 @@ def test_auto_remount_with_subpath():
...
-    14. In a 2 minutes retry loop:
-       Exec into the `pod-1`, run `ls /mnt/html`.
-       Verify the file `test_data.txt` exists.
+       Repeat these steps for 3 times.
+        1. Wait for volume `healthy`.
+        2. Verify the file checksum.
+    5. Update `numberOfReplicas` to 3.
+        1. Wait for replicas rebuilding finishes.
+    6. Delete one of the statefulset engine instance manager pod.
+        1. Verify the file checksum.
```
Comment (shuo-wu) [inline manager/integration/tests/test_ha.py:2052]: Wait for volume remount. Then verify the file checksum.

**Review by shuo-wu — CHANGES_REQUESTED** [review]
The test implementation part LGTM

**Comment by c3y1huang** [comment]
@shuo-wu thanks for the review, all addressed.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**